### PR TITLE
t: Add tests for power_action_utils

### PR DESCRIPTION
**Pull request by okurz**
Inspired by https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/9702

### CODE REVIEW

**Comment by okurz** [comment]
@foursixnine @SergioAtSUSE do you think tests like these are helpful? At least the test is clearly able to reproduce issues as caused by https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/9598

**Comment by foursixnine** [comment]
Useful is one thing, but feasible is a complete separate subject.

File: t/04_power_action_utils.t [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+set_var('BACKEND', 'spvm');
+my @calls;
+my $mock = Test::MockModule->new('power_action_utils');
+$mock->redefine(console => sub {
+        push @calls, @_;
+        Test::MockObject->new()->set_true('kill_ssh')->set_true('disable_vnc_stalls')->set_true('stop_serial_grab');
+});
+is prepare_system_shutdown, undef, 'prepare_system_shutdown for spvm is fine';
+is_deeply \@calls, ['root-ssh'], 'root-ssh console accessed';
+@calls = [];
```
Comment (foursixnine) [inline t/04_power_action_utils.t]: I'd separate asserts from variable assignments and test data stuff.

Comment (okurz) [reply t/04_power_action_utils.t]: What do you mean with "separate"? Add blank lines? Normally I see them used pretty arbitrarily so I better avoid them at all :) Or do you mean something different?

**Comment by okurz** [comment]
> Useful is one thing, but feasible is a complete separate subject.

With "useful" I also mean: Would it be feasible to ask contributors to provide tests in a similar fashion whenever they do (bigger) changes in lib/?

File: t/04_power_action_utils.t (hunk 1 above)
Comment (foursixnine) [reply t/04_power_action_utils.t]: Yes, blank lines

**Comment by foursixnine** [comment]
> > Useful is one thing, but feasible is a complete separate subject. 
> With "useful" I also mean: Would it be feasible to ask contributors to provide tests in a similar fashion whenever they do (bigger) changes in lib/?

That's what I meant with feasible :). But I'd say no, or at least not until a decent amount of tests are added

**Comment by okurz** [comment]
ok. Waiting for feedback from others.

**Comment by foursixnine** [comment]
@okurz 

> ok. Waiting for feedback from others.

Where's the feedback from others?

**Comment by okurz** [comment]
> Where's the feedback from others?

not yet here

**Pull request merged**